sources/packages/backend/apps/api/src/route-controllers/cas-supplier/_tests_/cas-supplier.aest.controller.retryCASSupplier.e2e-spec.ts

```diff
@@ -0,0 +1,106 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../testHelpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  saveFakeCASSupplier,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { SupplierStatus } from "@sims/sims-db";
+
+describe("CASSupplierAESTController(e2e)-retryCASSupplier", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it("Should throw Unprocessable Entity Error when retry CAS supplier is requested for a student who already has a CAS supplier in Pending supplier verification status.", async () => {
+    // Arrange
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+    // Create a CAS Supplier for the student with Pending supplier verification status.
+    await saveFakeCASSupplier(
+      db,
+      { student },
+      {
+        initialValues: {
+          supplierStatus: SupplierStatus.PendingSupplierVerification,
+          status: "ACTIVE",
+        },
+      },
+    );
+    const endpoint = `/aest/cas-supplier/retry/${student.id}`;
```

Comment:
he controller should be defined as `/aest/cas-supplier/student/:studentId/retry`.